src/benchmarks/micro/libraries/System/ConvertHalfToSingle.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+using BenchmarkDotNet.Attributes;
+
+using MicroBenchmarks;
+
+namespace System
```

Comment:
nit: These should instead go under the https://github.com/dotnet/performance/tree/main/src/benchmarks/micro/libraries/System.Runtime folder and follow the same general conventions as `Perf.Double.cs` and `Perf.Single.cs`

This includes adding the required license header, matching the general namespace, and matching the general other conventions used for passing in arguments/etc

We would likewise ideally not write our own loop here but rather let BDN handle doing the looping itself unless absolutely required. -- I don't believe its required here due to the innate overhead in doing `Half<->Single` conversions

Reply:
Thank you for your advice about folder hierarchy.
However, input permutation is absolutely necessary.
If we don't permute the input at all, the previous method can greatly take advantage of branch prediction.

Reply:
Benchmark.NET is design to appropriately measure things here and technology such as DPGO will itself optimize based on branches.

If there are multiple interesting paths then we should simply ensure the most interesting ones are covered by the benchmark as separate runs.